packages/govern-create/utils/etherscan.ts

```diff
@@ -0,0 +1,77 @@
+import HRE from 'hardhat'
+import fs from 'fs'
+import { file } from 'tmp-promise'
+
+export const SUPPORTED_ETHERSCAN_NETWORKS = ['mainnet', 'rinkeby']
+
+function delay(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+export const verifyContract = async (
+  address: string,
+  constructorArguments: any[]
+) => {
+  const currentNetwork = HRE.network.name
+
+  if (!process.env.ETHERSCAN_KEY) {
+    throw Error('Missing process.env.ETHERSCAN_KEY.')
+  }
+  if (!SUPPORTED_ETHERSCAN_NETWORKS.includes(currentNetwork)) {
+    throw Error(
+      `Current network ${currentNetwork} not supported. Please change to one of the next networks: ${SUPPORTED_ETHERSCAN_NETWORKS.toString()}`
+    )
+  }
+
+  try {
+    console.warn(
+      'Delaying Etherscan verification due their API can not find newly deployed contracts'
+    )
+    const msDelay = 3000
+    const times = 30
+    // Write a temporal file to host complex parameters for hardhat-etherscan https://github.com/nomiclabs/hardhat/tree/master/packages/hardhat-etherscan#complex-arguments
+    const { fd, path, cleanup } = await file({
+      prefix: 'verify-params-',
+      postfix: '.js',
+    })
+    fs.writeSync(
+      fd,
+      `module.exports = ${JSON.stringify([...constructorArguments])};`
+    )
+
+    const params = {
+      address: address,
+      constructorArgs: path,
+    }
+    await runTaskWithRetry('verify', params, times, msDelay, cleanup)
+  } catch (error) {}
```

Comment:
What is the rationale to hide this error?

Reply:
This was a snipped of code I forked from other project. As I understood the code was to prevent the hardhat `script/task/deploy` to stop execution because this error.